geomesa-spark/geomesa-spark-converter/src/main/scala/org/locationtech/geomesa/spark/converter/ConverterSpatialRDDProvider.scala

```diff
@@ -43,14 +44,15 @@ import scala.util.control.NonFatal
 class ConverterSpatialRDDProvider extends SpatialRDDProvider with LazyLogging {
 
   import ConverterSpatialRDDProvider.{ConverterKey, FeatureNameKey, InputFilesKey, SftKey}
+  import org.locationtech.geomesa.spark.TypeDefault._
 
   override def canProcess(params: util.Map[String, Serializable]): Boolean =
     params.containsKey(ConverterKey) && params.containsKey(SftKey) && params.containsKey(InputFilesKey)
 
-  override def rdd(conf: Configuration,
-                   sc: SparkContext,
-                   params: Map[String, String],
-                   query: Query): RDD[SimpleFeature] = {
+  override def rdd[T : ClassTag](conf: Configuration,
+                                 sc: SparkContext,
+                                 params: Map[String, String],
+                                 query: Query)(implicit default: T := SimpleFeature): RDD[T] = {
     val sftName = params.get(FeatureNameKey).orElse(Option(query.getTypeName)).orNull
```

Comment:
we should validate that the classtag is `SimpleFeature` up front - anything else will blow up down in the `.asInstanceOf` i believe